subsys/net/ip/tcp.c

```diff
@@ -1230,3 +1231,71 @@ struct net_buf *net_tcp_set_chksum(struct net_pkt *pkt, struct net_buf *frag)
 
 	return frag;
 }
+
+int tcp_parse_opts(struct net_pkt *pkt, int opt_totlen, u16_t *mss)
```

Comment:
Better would be to give also the option we are interested in like NET_TCP_MSS_OPT as a parameter, and then return the value according to wanted option.

Reply:
I don't think that would scale or follow any realistic processing routine. Realistically, we will always parse options once, and get all the interesting options during this parse too. So, I had an idea to introduce `struct tcp_options`, but it would contain only MSS now. So, I considered that over-engineering and applied YAGNI - indeed, it doesn't make sense to remove unused things in one place (https://github.com/zephyrproject-rtos/zephyr/issues/4653#issuecomment-340829440) and add unused stuff in another.

Reply:
Your "struct tcp_options" sounds like a good plan even if it would contain only MSS for now. Now the function is quite misleading as it only parses mss but you call it tcp_parse_opts(). I did not understand your last sentence, what we are removing and adding?

Reply:
> I did not understand your last sentence, what we are removing and adding?

What I meant that in #4653, an used structure field being reported, and e.g. @lpereira's suggestion was "remove it". Then we could follow that idea and not add fields/structures unless really needed.

Anyway, if you find that introducing `struct tcp_options` would deconfuse this patch, let me just do it.